src/Sulu/Component/Content/Types/Block/ScheduleBlockVisitor.php

```diff
@@ -0,0 +1,116 @@
+        $this->cacheLifetimeRequestEnhancer = $cacheLifetimeRequestEnhancer;
+    }
+
+    public function visit(BlockPropertyType $block): ?BlockPropertyType
+    {
+        $blockPropertyTypeSettings = $block->getSettings();
+
+        if (!\is_array($blockPropertyTypeSettings)
+            || !isset($blockPropertyTypeSettings['schedules_enabled'])
+            || !$blockPropertyTypeSettings['schedules_enabled']
+            || !isset($blockPropertyTypeSettings['schedules'])
+            || !\is_array($blockPropertyTypeSettings['schedules'])
+        ) {
+            return $block;
+        }
+
+        $now = $this->requestAnalyzer->getDateTime();
+        $nowTimestamp = $now->getTimestamp();
+
+        $returnBlock = false;
+
+        foreach ($blockPropertyTypeSettings['schedules'] as $schedule) {
+            switch ($schedule['type']) {
+                case 'fixed':
+                    $start = new \DateTime($schedule['start']);
+                    $end = new \DateTime($schedule['end']);
+
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($start->getTimestamp() - $nowTimestamp);
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($end->getTimestamp() - $nowTimestamp);
+
+                    if ($now >= $start && $now <= $end) {
+                        $returnBlock = true;
+                        continue 2;
+                    }
+                    break;
+                case 'weekly':
+                    $year = $now->format('Y');
+                    $month = $now->format('m');
+                    $day = $now->format('d');
+
+                    $start = new \DateTime($schedule['start']);
+                    $start->setDate($year, $month, $day);
+                    $end = new \DateTime($schedule['end']);
+                    $end->setDate($year, $month, $day);
+
+                    if ($end < $start) {
+                        $start->modify('-1 day');
+                    }
+
+                    $i = 0;
+                    do {
+                        if ($this->matchWeekday($start, $schedule)) {
+                            break;
+                        }
+
+                        $start->modify('+1 day');
+                        $end->modify('+1 day');
+                        ++$i;
+                    } while ($i < 7);
+
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($start->getTimestamp() - $nowTimestamp);
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($end->getTimestamp() - $nowTimestamp);
+
+                    if ($now >= $start && $now <= $end) {
+                        $returnBlock = true;
+                    }
+                    break;
+            }
+        }
+
+        return $returnBlock ? $block : null;
+    }
+
+    private function matchWeekday(\DateTime $datetime, $schedule)
+    {
+        if (!\is_array($schedule['days'])) {
+            return true;
```

Comment:
whats the reason for this? 🤔

Reply:
If the user has not selected any weekday, then the given times work for every weekday. Have added a comment.